code/datums/components/barber.dm

```diff
@@ -518,19 +522,45 @@ TYPEINFO_NEW(/datum/component/barber/shave)
 							src.new_AH.customization_second = new_hairstyle
 						if ("top")
 							src.new_AH.customization_third = new_hairstyle
-					src.preview.preview_thing.appearance = src.barbee.appearance
+					src.copy_clothes(src.barbee, src.preview.preview_thing)
 					src.preview.update_appearance(src.new_AH)
 
 				if("change_direction")
-					src.preview.preview_thing.dir = turn(src.preview.preview_thing.dir, params["direction"]) // Not using `update_appearance` to change dirs because it simply doesn't work. Why? Dunno.
+					src.preview.update_appearance(src.new_AH, src.new_AH.mutant_race, turn(src.preview.preview_thing.dir, params["direction"])) // Not using `update_appearance` to change dirs because it simply doesn't work. Why? Dunno.
 
 				if("reset")
 					src.new_AH.CopyOther(src.barbee.bioHolder.mobAppearance)
-					src.preview.preview_thing.appearance = src.barbee.appearance
+					src.copy_clothes(src.barbee, src.preview.preview_thing)
 					src.preview.update_appearance(src.new_AH)
 
 			return TRUE
 
+/datum/component/barber/proc/copy_clothes(var/mob/living/carbon/human/to_copy, var/mob/living/carbon/human/to_paste)
+	if (to_copy.wear_suit)
+		to_paste.wear_suit = new to_copy.wear_suit.type
+	if (to_copy.w_uniform)
+		to_paste.w_uniform = new to_copy.w_uniform.type
+	if (to_copy.shoes)
+		to_paste.shoes = new to_copy.shoes.type
+	if (to_copy.belt)
+		to_paste.belt = new to_copy.belt.type
+	if (to_copy.gloves)
+		to_paste.gloves = new to_copy.gloves.type
+	if (to_copy.glasses)
+		to_paste.glasses = new to_copy.glasses.type
+	if (to_copy.glasses)
+		to_paste.glasses = new to_copy.glasses.type
+	if (to_copy.head)
+		to_paste.head = new to_copy.head.type
+	if (to_copy.wear_id)
+		to_paste.wear_id = new to_copy.wear_id.type
+	if (to_copy.r_store)
+		to_paste.r_store = new to_copy.r_store.type
+	if (to_copy.l_store)
+		to_paste.l_store = new to_copy.l_store.type
+
+	to_paste.update_clothing()
+
```

Comment:
this is one massive memory leak, either don't replace them every time or at least `qdel` the old ones

Reply:
![image](https://user-images.githubusercontent.com/20713227/232809971-cc40d0c2-9dad-47c5-964a-51cca390768a.png)

Reply:
oh, right, shit i forgot that. that's one dumb mistake. sorry, i'll fix that.